### PR TITLE
Chore: adjust button, and text components

### PR DESCRIPTION
**Pull request by micbaumr**
# Description

Add button variants according to designs. 
Add text (text/display) variants according to desings.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/design-system/components/Button/Button.tsx [hunk 1]
```diff
@@ -20,18 +20,19 @@ import {
...
 import { Text } from '../Text'
+import { useHover } from '../Touchables/useHover'
 import { StyledProps } from '../types'
 
 export type ButtonProps = StyledProps &
   PressableProps & {
     title?: string
     variant?: ButtonVariant
-    size?: 'sm' | 'md' | 'lg'
+    size?: 'sm' | 'md' | 'lg' | 'xl' | '2xl'
```
Comment (MateuszRostkowski) [inline src/design-system/components/Button/Button.tsx:35]: Maybe we could ask designers to change `2xl` to `xxl` 🤔

File: src/design-system/components/Text/Text.tsx [hunk 2]
```diff
@@ -183,58 +203,71 @@ const RawText = memo(
...
+
+const generateDisplayVariant = (variant: DisplayVariant = 'MdRegular') =>
+  forwardRef<BaseText, TextProps>((props, ref) => {
+    return <Display {...{ ...props, ref, variant }} type="display" />
+  })
+
+Display['2xlBold'] = generateDisplayVariant('2xlBold')
+Display['2xlMedium'] = generateDisplayVariant('2xlMedium')
+Display['2xlRegular'] = generateDisplayVariant('2xlRegular')
+Display['2xlSemibold'] = generateDisplayVariant('2xlSemibold')
```
Comment (MateuszRostkowski) [inline src/design-system/components/Text/Text.tsx:250]: Maybe we could ask designers to change 2xl to xxl 🤔

Comment (micbaumr) [reply src/design-system/components/Text/Text.tsx:250]: Good idea! I will ask

Comment (micbaumr) [reply src/design-system/components/Text/Text.tsx:250]: Maybe, even if it could not be achieved, we could use this convention in codebase 🤔

**Pull request merged**